packages/true-time/src/index.ts

```diff
@@ -27,26 +40,42 @@ export class TrueTime {
     );
   }
 
+  async setServerTime() {
+    try {
+      const response = await fetch(this.#url);
+
+      if (response.ok && response.headers.has('date')) {
+        this.#serverTime = new Date(response.headers.get('date')!).getTime();
+        // eslint-disable-next-line no-restricted-syntax
+        this.#synced = Date.now();
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  }
+
   /**
    * Use this method to synchronize time with the server.
    *
    * @param url - server url
    */
   async synchronize() {
-    if (this.#synced) {
+    // Synchronize at max once every 1 000 000 miliseconds
+    // eslint-disable-next-line no-restricted-syntax
+    if (
+      (this.#synced && Math.abs(this.#synced - Date.now()) < 1_000_000) ||
+      (this.#synced &&
+        Math.abs(this.#initialDrift - this.currentDrift()) < 1_000)
```

Comment:
This "feels" like it is getting complicated, are we sure there are any benefits to not just replacing `performance.now()` with `Date.now()` and keeping this module simple?

Reply:
Then we'd have to re-implement performance.mark() with Date.now too... :/

Reply:
Yeah it is a bit unfortunate that that is mixed in here, makes this more complex than it should be :-(

Might need to think a bit how we can avoid that..